### ISSUE TITLE
Fix full changelog notes retrieval for unstable releases (using `ignoreUnstable=false`)

### ISSUE THREAD

**Issue by atorrescogollo** [renovatebot/renovate#15902]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.74.3

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When using `ignoreUnstable=false`, renovatebot will only fetch changelog notes from the last version. Debugging it, I found that it is filtered [in this lines](https://github.com/renovatebot/renovate/blob/32.74.3/lib/workers/repository/update/pr/changelog/releases.ts#L49-L54) because `config.ignoreUnstable` is not taken into account

### Relevant debug logs

N/A

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
If the upgrade is from stable to stable, I don't think we need to display release notes for the unstable/pre-releases which happened in-between. But if either from or to versions are unstable, we should show unstable release notes. However, not necessarily in every case.. for example if current version is 1.0.0, and new version is 1.2.0-pre.1 then we should display 1.1.0 plus all 1.2.0 unstable releases up to 1.2.0-pre.1. There would be no reason to show release notes of 1.1.0-alpha, beta or pre because they should be included in the 1.1.0 release notes.

**Comment by atorrescogollo** [comment]
My use case is not mixing versions with stable ones but using renovate with airbyte https://github.com/airbytehq/airbyte/releases, which publishes every release as unstable using the `-alpha` suffix.

If, for instance, I have `v0.39.0-alpha` set and the latest is `v0.39.10-alpha`, I won't see any release notes from `v0.39.1-alpha` to `v0.39.9-alpha`, but only the `v0.39.10-alpha` release notes

**Comment by rarkins** [comment]
I guess then if ignoreUnstable=false then you prefer to list every single release between from and to?

**Comment by atorrescogollo** [comment]
Exactly. I expected to have notes for each release from `v0.39.1-alpha` to `v0.39.10-alpha` (same as using stable since I'm unsetting `ignoreUnstable`), but only `v0.39.10-alpha` showed up.

**Comment by rarkins** [comment]
Ok I'm fine with that

**Comment by atorrescogollo** [comment]
Minimal reproduction created: https://github.com/atorrescogollo/renovate-15902-unstable-full-changelog/pull/1

As you can see, only the latest version changelog notes are published

**Comment by atorrescogollo** [comment]
Yes, maybe the cause is this issue https://github.com/renovatebot/renovate/pull/15897#issuecomment-1147331081 (pending to be reverted in https://github.com/renovatebot/renovate/pull/15909)? I had to git-reset the previous commit in the main branch (https://github.com/renovatebot/renovate/commit/4b6fc85146b6563f2da1633518c8c9c54afc2601). Otherwise, I was receiving `TypeError: Cannot read property 'map' of undefined`.

**Comment by rarkins** [comment]
The problematic has been reverted instead, to ensure it's fixed. Please merge from `main` branch and try again

**Comment by atorrescogollo** [comment]
Hey @MaronHatoum, was that the problem? Or do I need to do something else?

**Comment by MaronHatoum** [comment]
Hi @atorrescogollo 
I tried to reproduce the issue with the minimal reproduction you provided, but unfortunately, I was not able to, I got this log message from Renovate seems like the regex is not able to match the dependency:
```
DEBUG: Matched 1 file(s) for manager regex: airbyte.yaml (repository=MaronHatoum/renovate-15902-unstable-full-changelog)
DEBUG: Found 0 package file(s) (repository=MaronHatoum/renovate-15902-unstable-full-changelog)
DEBUG: Manager explicitly enabled in "enabledManagers" config, but found no results. Possible config error? (repository=MaronHatoum/renovate-15902-unstable-full-changelog)
       "manager": "regex"
 INFO: Dependency extraction complete (repository=MaronHatoum/renovate-15902-unstable-full-changelog)
       "baseBranch": "main",
       "stats": {"managers": {}, "total": {"fileCount": 0, "depCount": 0}}
```
would you have a suggestion for how to fix the regex?

**Comment by atorrescogollo** [comment]
Ok, I think it was the newline character. This was a copy-pasted fragment from a real repo and it contained a `\n` in the end, but the yaml doesn't have an extra newline. Just rebase your fork and I guess this should work. Tell me if not. Sorry about that.

**Comment by rarkins** [comment]
Please reproduce it in the hosted app so this can't happen again. Reproductions shouldn't be copy pastes for this reason

**Comment by atorrescogollo** [comment]
Actually, I tested running it locally. As you can see, the PR was created correctly: https://github.com/atorrescogollo/renovate-15902-unstable-full-changelog/pull/1

But I can't ensure if other changes in the codebase will trim the source file or change the behavior (not sure why this thing was causing an error with newer versions). The reproduction is working, as I [said before](https://github.com/renovatebot/renovate/issues/15902#issuecomment-1147465985), with the [latest working commit](https://github.com/renovatebot/renovate/commit/4b6fc85146b6563f2da1633518c8c9c54afc2601) at the date of the issue creation

**Comment by atorrescogollo** [comment]
Just tested with the latest renovate commit in main branch. The initial approach works but does work when replacing `\n` with `$`. So that was not the reason it was failing

**Comment by atorrescogollo** [comment]
Also tested with renovate hosted app: https://github.com/atorrescogollo/renovate-15902-unstable-full-changelog/issues/4
